cmd/skopeo/sync.go

```diff
@@ -489,7 +556,7 @@ func imagesToCopy(source string, transport string, sourceCtx *types.SystemContex
 			return descriptors, err
 		}
 		for registryName, registryConfig := range cfg {
-			if len(registryConfig.Images) == 0 && len(registryConfig.ImagesByTagRegex) == 0 {
+			if len(registryConfig.Images) == 0 && len(registryConfig.ImagesByTagRegex) == 0 && len(registryConfig.ImagesBySemver) == 0 {
```

Comment:
This is pretty far from the actual uses of those values, and risks getting out of sync — moving this whole check into `imagesToCopyFromRegistry` would be nice.